upup/pkg/fi/cloudup/awstasks/securitygroup.go

```diff
@@ -96,26 +96,22 @@ func (e *SecurityGroup) Find(c *fi.Context) (*SecurityGroup, error) {
 
 func (e *SecurityGroup) findEc2(c *fi.Context) (*ec2.SecurityGroup, error) {
 	cloud := c.Cloud.(awsup.AWSCloud)
-
-	var vpcID *string
-	if e.VPC != nil {
-		vpcID = e.VPC.ID
-	}
-
-	if vpcID == nil {
-		return nil, nil
-	}
-
 	request := &ec2.DescribeSecurityGroupsInput{}
 
 	if fi.StringValue(e.ID) != "" {
+		// Find by ID.
 		request.GroupIds = []*string{e.ID}
-	} else {
+
+	} else if fi.StringValue(e.Name) != "" && e.VPC != nil {
```

Comment:
The lines removed above were also validating that `e.VPC.ID` is not null. Was this an invalid case, or do you need to extend your if check here?

Reply:
@KashifSaadat Thanks! Good catch, [fixed](https://github.com/kubernetes/kops/pull/6030/commits/fdd8a3084142f2700c4454f5cd60b6fa027c7123).